pkgs/development/compilers/sapmachine/11.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchurl, unzip }:
+
+stdenv.mkDerivation rec {
+
+  pname = "sapmachine";
+
+  version = "11.0.5";
+
+  sha256_linux = "f8b849bbd044acfd38972987305bd4c8bf5706a794ec12b51f1ed747a492ca55";
+  sha256_darwin = "31fcd79060ca5c62be9866450cf501361c0e478f3418c366a136f63a7a530059";
+
+  platform = if stdenv.isDarwin then "osx" else "linux";
+  sha256 = if stdenv.isDarwin then sha256_darwin else sha256_linux;
+  homepath = if stdenv.isDarwin then "Contents/Home" else ".";
+
+  src = fetchurl {
+    url = "https://github.com/SAP/SapMachine/releases/download/sapmachine-${version}/sapmachine-jdk-${version}_${platform}-x64_bin.tar.gz";
+    sha256 = sha256;
+  };
+
+  buildInputs = [ unzip];
+  
+  installPhase = ''
+    mkdir -p $out
+    cp -r ./$homepath/* "$out/"
+
+    # Set JAVA_HOME
+    mkdir -p $out/nix-support
+    cat <<EOF >> $out/nix-support/setup-hook
+    if [ -z "\$JAVA_HOME" ]; then export JAVA_HOME=$out; fi
+    EOF
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = https://sap.github.io/SapMachine/;
+    license = licenses.gpl2;
```

Comment:
The gpl license variants without the `Plus` or `Only` suffix are deprecated, because they do not specify the license well enough. Use `Plus` if the project has "or later" sections in the actual source code files (not the license). Otherwise, use the `Only` suffix.